docs/0.5/howto/using_circuit_templates.md

```diff
@@ -0,0 +1,367 @@
+# Using Circuit Templates
+
+<!--
+  Copyright 2018-2020 Cargill Incorporated
+  Licensed under Creative Commons Attribution 4.0 International License
+  https://creativecommons.org/licenses/by/4.0/
+-->
+
+This guide explains how to use the `circuit-template` feature of Splinter,
+including explaining the concept and format of a circuit template file and
+how to use the `circuit-template` feature when developing an application.
+
+## Overview
+
+A circuit template contains a set of `rule`s which partially define a new circuit.
+This can make circuit creation substantially easier for users as only a minimal
+amount of information needs be provided--the remainder is defined in the circuit
+template.
+
+The following is an example circuit template YAML file specific to the Gameroom
+application:
+
+```
+version: v1
+args:
+    - name: ADMIN_KEYS
+      required: false
+      default: $(SIGNER_PUB_KEY)
+      description: >-
+        Public keys used to verify transactions in the scabbard service
+    - name: NODES
+      required: true
+      description: "List of node IDs"
+    - name: SIGNER_PUB_KEY
+      required: false
+      description: "Public key of the signer"
+    - name: GAMEROOM_NAME
+      required: true
+      description: "Name of the gameroom"
+rules:
+    set-management-type:
+        management-type: "gameroom"
+    create-services:
+        service-type: 'scabbard'
+        service-args:
+        - key: 'admin_keys'
+          value: [$(ADMIN_KEYS)]
+        - key: 'peer_services'
+          value: '$(ALL_OTHER_SERVICES)'
+        first-service: 'a000'
+    set-metadata:
+        encoding: json
+        metadata:
+            - key: "scabbard_admin_keys"
+              value: ["$(ADMIN_KEYS)"]
+            - key: "alias"
+              value: "$(GAMEROOM_NAME)"
+```
+
+There are three main sections to the circuit template, `version`, `arg`s, and
+`rule`s.
+
+## Version
+
+This describes the version of circuit the template is targeting.
+
+## Args
+
+`Arg`s are the arguments which affect the circuit. An argument has a few
+attributes. The following is from the example YAML above:
+
+```
+- name: SIGNER_PUB_KEY
+  required: false
+  description: "Public key of the signer"
+```
+
+This shows the definition of the `SIGNER_PUB_KEY` argument value. An argument
+value can be specified as `required` or not, as well as a short description
+string. This argument will be assigned a value when the circuit is being
+created. Argument values may also be assigned an optional default value, as can
+be seen in the next example. The following shows the definition of the
+`ADMIN_KEYS` argument value.
+
+```
+- name: ADMIN_KEYS
+  required: false
+  default: $(SIGNER_PUB_KEY)
+  description: >-
+    Public keys used to verify transactions in the scabbard service
+```
+
+The value of `SIGNER_PUB_KEY` is used as the default value of the `ADMIN_KEYS`
+argument.
+
+## Rules
```

Comment:
In this section, I would remove much of the code quotes, except when directly referring to the `rules` section.  Each individual thing is a "rule", but it is not a code term that anyone uses when constructing a template.